src/lib/chawathe98_matcher.rs

```diff
@@ -46,35 +46,2262 @@
 
 use std::fmt::Debug;
 
-use ast::Arena;
-use matchers::{MappingStore, MatchTrees};
+use ast::{Arena, FromNodeId, NodeId, ToNodeId};
 
-/// The Chawathe98Config matcher needs no configuration.
+use std::cell::RefCell;
+use std::collections::HashSet;
+use std::cmp;
+
+/// Not needed by matching algorithms, but useful for debugging.
+#[derive(Debug, Clone, Eq, Hash, PartialEq)]
+pub enum EdgeType {
+    /// Check the type of edge update which is update
```

Comment:
These "Check the type" comments in the enum are not so useful. If you want to keep them they need to be rephrased a bit (e.g. "Edge corresponds to an UPDATE operation").